# Issue 15453: Wrap IML calls in sig_on()

archive/issues_015216.json:
```json
{
    "body": "In `devel/sage/sage/matrix/matrix_integer_dense.pyx`, wrap IML calls in `_solve_iml()` in `sig_on()`/`sig_off()`.\n\nThe patch also changes the exception type for non-square matrices from `ArithmeticError` to `ValueError`, since that's just a limitation of the implementation, it's not a mathematical obstruction.\n\nDepends on #13311\n\n**CC:**  @malb\n\n**Reviewer:** Martin Albrecht\n\n**Author:** Jeroen Demeyer\n\n**Merged:** sage-5.13.beta5\n\nIssue created by migration from https://trac.sagemath.org/ticket/15453\n\n",
    "closed_at": "2013-12-05T08:01:26Z",
    "created_at": "2013-11-25T21:17:40Z",
    "labels": [
        "component: linear algebra",
        "bug"
    ],
    "milestone": "https://github.com/sagemath/sage/milestones/sage-5.13",
    "title": "Wrap IML calls in sig_on()",
    "type": "issue",
    "url": "https://github.com/sagemath/sage/issues/15453",
    "user": "https://github.com/jdemeyer"
}
```
In `devel/sage/sage/matrix/matrix_integer_dense.pyx`, wrap IML calls in `_solve_iml()` in `sig_on()`/`sig_off()`.

The patch also changes the exception type for non-square matrices from `ArithmeticError` to `ValueError`, since that's just a limitation of the implementation, it's not a mathematical obstruction.

Depends on #13311

**CC:**  @malb

**Reviewer:** Martin Albrecht

**Author:** Jeroen Demeyer

**Merged:** sage-5.13.beta5

Issue created by migration from https://trac.sagemath.org/ticket/15453





---

archive/issue_events_137404.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-11-25T21:57:58Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137404"
}
```



---

archive/issue_comments_194875.json:
```json
{
    "body": "**Dependencies:** #13311",
    "created_at": "2013-11-25T21:57:58Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/15453#issuecomment-194875",
    "user": "https://github.com/jdemeyer"
}
```

**Dependencies:** #13311



---

archive/issue_events_137405.json:
```json
{
    "actor": "https://github.com/malb",
    "created_at": "2013-11-26T12:14:32Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137405"
}
```



---

archive/issue_events_137406.json:
```json
{
    "actor": "https://github.com/malb",
    "created_at": "2013-11-26T12:14:32Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "needs work",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137406"
}
```



---

archive/issue_comments_194876.json:
```json
{
    "body": "**Reviewer:** Martin Albrecht",
    "created_at": "2013-11-26T12:14:32Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/15453#issuecomment-194876",
    "user": "https://github.com/malb"
}
```

**Reviewer:** Martin Albrecht



---

archive/attachments_021137.json:
```json
{
    "asset_content_type": "application/octet-stream",
    "asset_name": "15453_iml_sig_on.patch",
    "asset_url": "tarball://root/attachments/ticket15453/15453_iml_sig_on.patch",
    "created_at": "2013-11-26T12:14:32Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "attachment",
    "url": "https://github.com/sagemath/sage/files/ticket15453/15453_iml_sig_on.patch",
    "user": "https://github.com/malb"
}
```



---

archive/issue_comments_194877.json:
```json
{
    "body": "<a id='comment:2'></a>\n**Attachment:** [15453_iml_sig_on.patch](https://github.com/sagemath/sage/files/ticket15453/15453_iml_sig_on.patch)\n\npatch looks good and passes test. However, can you document the fact that you changed the exception in the ticket's description. After that, it's an automatic positive review.",
    "created_at": "2013-11-26T12:14:32Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/15453#issuecomment-194877",
    "user": "https://github.com/malb"
}
```

<a id='comment:2'></a>
**Attachment:** [15453_iml_sig_on.patch](https://github.com/sagemath/sage/files/ticket15453/15453_iml_sig_on.patch)

patch looks good and passes test. However, can you document the fact that you changed the exception in the ticket's description. After that, it's an automatic positive review.



---

archive/issue_events_137407.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-11-26T12:21:07Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "needs work",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137407"
}
```



---

archive/issue_events_137408.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-11-26T12:21:07Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137408"
}
```



---

archive/issue_comments_194878.json:
```json
{
    "body": "**Description changed:**\n``````diff\n--- \n+++ \n@@ -1 +1,3 @@\n In `devel/sage/sage/matrix/matrix_integer_dense.pyx`, wrap IML calls in `_solve_iml()` in `sig_on()`/`sig_off()`.\n+\n+The patch also changes the exception type for non-square matrices from `ArithmeticError` to `ValueError`, since that's just a limitation of the implementation, it's not a mathematical obstruction.\n``````\n",
    "created_at": "2013-11-26T12:21:07Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/15453#issuecomment-194878",
    "user": "https://github.com/jdemeyer"
}
```

**Description changed:**
``````diff
--- 
+++ 
@@ -1 +1,3 @@
 In `devel/sage/sage/matrix/matrix_integer_dense.pyx`, wrap IML calls in `_solve_iml()` in `sig_on()`/`sig_off()`.
+
+The patch also changes the exception type for non-square matrices from `ArithmeticError` to `ValueError`, since that's just a limitation of the implementation, it's not a mathematical obstruction.
``````




---

archive/issue_comments_194879.json:
```json
{
    "body": "**Merged:** sage-5.13.beta5",
    "created_at": "2013-12-05T08:01:26Z",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/15453#issuecomment-194879",
    "user": "https://github.com/jdemeyer"
}
```

**Merged:** sage-5.13.beta5



---

archive/issue_events_137409.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-12-05T08:01:26Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137409"
}
```



---

archive/issue_events_137410.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2013-12-05T08:01:26Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/15453",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/15453#event-137410"
}
```
